src/goto-instrument/unwindset.cpp

```diff
@@ -60,26 +79,28 @@ void unwindsett::parse_unwindset(const std::string &unwindset)
   }
 }
 
-optionalt<unsigned>
-unwindsett::get_limit(const irep_idt &loop_id, unsigned thread_nr) const
+optionalt<unsigned> unwindsett::get_limit(
+  const irep_idt &loop_id,
+  unsigned thread_nr,
+  bool is_iteration) const
```

Comment:
Please don't use a boolean flag for this, they have too many overloaded meanings already. Either pass in the applicable global limit directly (+ possibly refactor this into `get_loop_limit` and `get_recursion_limit`?) or use an enum like

```
enum class unwind_kindt { loop, recursion };
```